scipy/linalg/_basic.py

```diff
@@ -141,11 +145,37 @@ def solve(a, b, lower=False, overwrite_a=False,
     array([ True,  True,  True], dtype=bool)
 
     """
+    xp = array_namespace(a, b)
+    if check_finite:
+        a = _asarray(a, check_finite=True, xp=xp)
+        b = _asarray(b, check_finite=True, xp=xp)
+    if is_numpy(xp):
+        return _solve(a, b, lower=lower, overwrite_a=overwrite_a,
+                      overwrite_b=overwrite_b, assume_a=assume_a, transposed=transposed)
+    unsupported_args = {
+        'lower': lower,
+        'assume_a': assume_a != 'gen',
+        'transposed': transposed
+    }
+    if any(unsupported_args.values()):
+        xp_unsupported_args(unsupported_args)
+    if hasattr(xp, 'linalg'):
```

Comment:
Is it possible for `xp` to have `linalg` but `linalg` not to have `solve`?

Reply:
Yes, but we don't want to support such a library. Ref: https://data-apis.org/array-api/latest/extensions/index.html

> Extension module implementors must aim to provide all functions and other public objects in an extension. The rationale for this is that downstream usage can then check whether or not the extension is present (using hasattr(xp, 'extension_name') should be enough), and can then assume that functions are implemented. This in turn makes it also easy for array-consuming libraries to document which array libraries they support - e.g., “all libraries implementing the array API standard and its linear algebra extension”.